dark/sam.py

```diff
@@ -1262,3 +1260,233 @@ def getReferenceInfo(
+        @return: A C{list} of sorted C{ReferenceReads} instances.
+        """
+        totalReads = len(self.readIds)
+        result = [
+            "Found a total of %d read%s (%d mapped, %d unmapped) mapping "
+            "against %d of %d reference%s."
+            % (
+                totalReads,
+                "" if totalReads == 1 else "s",
+                len(self.mapped),
+                len(self.unmapped),
+                len(self.references),
+                len(self.referenceLengths),
+                "" if len(self.referenceLengths) == 1 else "s",
+            )
+        ]
+
+        if not self.references:
+            result.append("No reads were mapped to any reference.")
+            return "\n".join(result)
+
+        cumulativeReadIds: set[str] = set()
+
+        if self.sortedBy != sortBy:
+            self.sortedReferences = self.sortReferences(sortBy)
+
+        for count, reference in enumerate(self.sortedReferences, start=1):
+            readIds = reference.readIds
+            readCount = len(readIds)
+            if readCount == 0 and excludeZeroes:
+                continue
+            newReadCount = len(readIds - cumulativeReadIds)
+            if newReadCount == 0 and excludeIfNoAdditional:
+                continue
+            cumulativeReadIds.update(readIds)
+            result.append(
+                "\nReference %d: %s (%d nt):\n"
+                "  Genome coverage: %.2f%%\n"
+                "  Overall reads mapped to the reference: %s\n"
+                "    Non-duplicates: %s, Duplicates: %s, QC fails: %s\n"
+                "    Primary: %s, Secondary: %s, Supplementary: %s\n"
+                "    Reads not matching any reference above: %s\n"
+                "  Previously unmatched reads for this reference: %s"
+                % (
+                    count,
+                    reference.id_,
+                    reference.length,
+                    reference.coverage() * 100.0,
+                    pct(readCount, totalReads),
+                    pct(len(reference.nonDuplicate), readCount),
+                    pct(len(reference.duplicate), readCount),
+                    pct(len(reference.qcFail), readCount),
+                    pct(len(reference.primary), readCount),
+                    pct(len(reference.secondary), readCount),
+                    pct(len(reference.supplementary), readCount),
+                    pct(newReadCount, totalReads),
+                    pct(newReadCount, readCount),
+                )
+            )
+
+        return "\n".join(result)
+
+    def sortReferences(self, sortBy: str = "count"):
+        """
+        Sort the references.
+
+        @param sortBy: A C{str} indicating how to sort. Use 'count' to sort by
+            decreasing read count (i.e., the reference with the highest number
+            of matching reads comes first), or 'coverage' to sort by decreasing
+            genome coverage, or 'name' to sort by reference name."
```

Comment:
Is there a stray `"` at the end of the line?

Reply:
Yes, and I even found another one :-) Thanks!